wiki/Test-infrastructure.md

```diff
@@ -0,0 +1,73 @@
+Test infrastructure
+---
+
+### Running tests
+
+If you just want to run all available tests please use `./run_tests.py`. This is
+the command that is used by continuous integration. Some useful arguments are
+listed below:
+
+* `--infinite` - keep testing until some error is found.
+* `--non-interactive` - do not run interactive gdb session if tests fail.
+* `--thorough` - generate much more test seeds. Testing will take much more time.
+
+For greater control you can pass kernel arguments that control test runs. That
+includes passing `seed` which is used to generate tests permutation, which
+proved to be useful. Some useful kernel parameters for test run control are
+listed below:
+
+* `test=TEST` - Requests the kernel to run the specified test.
+  `test=user_{name}` to run single user test, `test={name}` to run single kernel
+  test.
+  You can specify multiple tests by separating their names with commas (without spaces),
+  e.g. `test=test1,test2,test3`
+* `test=all` - Runs a number of tests one after another, and reports success
+  only when all of them passed.
+* `seed=UINT` - Sets the RNG seed for shuffling the list of test when using
+  `test=all`.
+* `repeat=UINT` - Specifies the number of (shuffled) repetitions of each test
+  when using `test=all`.
+  When running user-specified tests (i.e. when `test` is not equal to `all`),
+  `repeat` specifies the number of times each test will be run. For example,
+  `test=test1,test2 repeat=4` will run `test1` 4 times and then run `test2` 4 times.
+
+### Programming tests
+
+##### Kernel tests
+Located in `$(TOPDIR)/tests`.
```

Comment:
Is `$(TOPDIR)` equal to `/sys`?

Reply:
No idea, but `TOPDIR` doesn't look like it, so I'll change it.